compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Optics.hs

```diff
@@ -213,7 +213,7 @@ instance MonoTraversable ModuleRef InterfaceInstanceMethod
 
 instance MonoTraversable ModuleRef FeatureFlags
 instance MonoTraversable ModuleRef Module
-instance MonoTraversable ModuleRef PackageMetadata
+instance MonoTraversable ModuleRef PackageMetadata where monoTraverse _ = pure
```

Comment:
???

Reply:
ah this is a funny one. [The `MonoTraversable` class](https://github.com/digital-asset/daml/blob/537447d92c7975d8ec853c98f25b1b208eebdee2/libs-haskell/da-hs-base/src/Control/Lens/MonoTraversal.hs#L20), like its name says, works like `Traversable` but limited to fields that don't change type

```haskell
  monoTraverse :: (MonoTraversable e te, Applicative f) => (e -> f e) -> te  -> f  te
  traverse     :: (    Traversable   t , Applicative f) => (a -> f b) -> t a -> f (t b)
```

AFAIK, we only use this to collect all the `ModuleRef`s in a package/module, to know what packages it depends on. Our definition of `MonoTraversable` has a default implementation in terms of GHC Generics which for product types simply composes the instances for each side of the product. The empty `instance` declaration meant that this default implementation was used for `PackageMetadata` (resulting in a traversal that just ignores the given function, since none of the fields of `PackageMetadata` contain `ModuleRef`s). With the new field, the generic implementation would require an instance `MonoTraversable ModuleRef PackageId`, which I'm not comfortable adding, so instead I replaced this implementation with an explicit `monoTraverse _ = pure` since we can say that the `PackageMetadata` will not (for now) bring any dependencies.